machine_learning/local_weighted_learning/local_weighted_learning.py

```diff
@@ -0,0 +1,133 @@
+# Required imports to run this file
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+# weighted matrix
+def weighted_matrix(point: np.mat, training_data_x: np.mat, bandwidth: float) -> np.mat:
+    """
+    Calculate the weight for every point in the
+    data set. It takes training_point , query_point, and tau
+    Here Tau is not a fixed value it can be varied depends on output.
+    tau --> bandwidth
+    xmat -->Training data
+    point --> the x where we want to make predictions
+    >>> weighted_matrix(np.array([1., 1.]),np.mat([[16.99, 10.34], [21.01,23.68],
+                        [24.59,25.69]]), 0.6)
+    matrix([[1.43807972e-207, 0.00000000e+000, 0.00000000e+000],
+            [0.00000000e+000, 0.00000000e+000, 0.00000000e+000],
+            [0.00000000e+000, 0.00000000e+000, 0.00000000e+000]])
+    """
+    # m is the number of training samples
+    m, n = np.shape(training_data_x)
+    # Initializing weights as identity matrix
+    weights = np.mat(np.eye((m)))
+    # calculating weights for all training examples [x(i)'s]
+    for j in range(m):
+        diff = point - training_data_x[j]
+        weights[j, j] = np.exp(diff * diff.T / (-2.0 * bandwidth ** 2))
+    return weights
+
+
+def local_weight(
+    point: np.mat, training_data_x: np.mat, training_data_y: np.mat, bandwidth: float
+) -> np.mat:
+    """
+    Calculate the local weights using the weight_matrix function on training data.
+    Return the weighted matrix.
+    >>> local_weight(np.array([1., 1.]),np.mat([[16.99, 10.34], [21.01,23.68],
+                     [24.59,25.69]]),np.mat([[1.01, 1.66, 3.5]]), 0.6)
+    matrix([[0.    ],
+            [0.0625]])
+    """
+    weight = weighted_matrix(point, training_data_x, bandwidth)
+    W = (training_data_x.T * (weight * training_data_x)).I * (
+        training_data_x.T * weight * training_data_y.T
+    )
+
+    return W
+
+
+def local_weight_regression(
+    training_data_x: np.mat, training_data_y: np.mat, bandwidth: float
+) -> np.mat:
+    """
+    Calculate predictions for each data point on axis.
+    >>> local_weight_regression(np.mat([[16.99, 10.34], [21.01,23.68],
+                                [24.59,25.69]]),np.mat([[1.01, 1.66, 3.5]]), 0.6)
+    array([1.2925    , 1.6589624 , 3.50142395])
+    """
+    m, n = np.shape(training_data_x)
+    ypred = np.zeros(m)
+
+    for i, item in enumerate(training_data_x):
+        ypred[i] = item * local_weight(
+            item, training_data_x, training_data_y, bandwidth
+        )
+
+    return ypred
+
+
+def load_data(dataset_name: str, cola_name: str, colb_name: str) -> np.mat:
+    """
+    Function used for loading data from the seaborn splitting into x and y points
+    """
```

Comment:
Let's silence @algorithms-keeper
```suggestion
    >>> pass  # This function has no doctest.
    """
```